crates/proto/src/udp/udp_client_stream.rs

```diff
@@ -70,6 +72,12 @@ impl<P> UdpClientStreamBuilder<P> {
         self
     }
 
+    /// Configures that OS should provide the ephemeral port, not the Hickory DNS
+    pub fn with_os_ephemeral_port(mut self, request_udp_port_from_os: bool) -> Self {
```

Comment:
I think it would be more clear to use one name for this config flag.

Reply:
I'd propose we call it `os_port_selection` and use consistent naming for both the fields and the methods. For builder methods, I don't think we need the argument since it's just a bool. The docstring should have some context about the tradeoff the caller is making -- explain what the default is and why and what reasons people might have to disable the default.